### PR TITLE
add stricter linting / formatting rules to web

### PR DESCRIPTION
**Pull request by brpins**
## Describe your changes

Add dedicated eslint-plugin-pins package
Adopt eslint extends paradigm instead of commonjs imports
Tweak prettier rules

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-  "printWidth": 156,
+  "printWidth": 100,
   "singleQuote": true,
-  "tabWidth": 4,
+  "tabWidth": 2,
```
[the remark this answers is not in the document]
Comment (brpins) [reply .prettierrc:4]: Ok - will do.

File: .prettierrc [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-  "printWidth": 156,
+  "printWidth": 100,
```
[the remark this answers is not in the document]
Comment (brpins) [reply .prettierrc:2]: Yes, it's true that sometimes you will have those couple of characters that can force a wrap (that's partly why tabWidth matters). However, these are edge cases, and if they're not and they are happening regularly, it's probably an issue with the code being written less than optimally (either too much nesting, or the line is too complex). Personally, I only ever run into this issue where variable names are excessively long.

The "solution" of longer lines is worse: The edge cases can be addressed by the developer through refactoring if they care that much, but wrapping being prevented in places where it was needed (such as chaining) cannot be fixed as prettier has the final say.

[the remark this answers is not in the document]
Comment (radum) [reply .prettierrc:2]: Uuuu first one we have to agree to disagree 😁. But it is your codebase now so happy with the changes.

File: .editorconfig [hunk 3]
```diff
@@ -8,7 +8,7 @@ root = true
 
 # Change these settings to your own preference
 indent_style = tab
-indent_size = 4
+indent_size = 2
```
Comment (radum) [inline .editorconfig:11]: Doesn't this change the tab_width also making the tab char smaller?

**Review by radum — DISMISSED** [review]

**Review by radum — APPROVED** [review]

File: .editorconfig (hunk 3 above)
Comment (brpins) [reply .editorconfig]: reverted

**Pull request merged**